### PR TITLE
[Enhancement]Support iceberg time type

### LINKED ISSUE #35231: starrocks mysql cli could not map iceberg `TIME` data type correctly
hi team, starrocks cannot handle iceberg time data type correctly.
I am creating a table in trino with a data type of `TIME(6)`, which mapping to `TIME` data type in iceberg.
then I could not load this table in starrocks due to unknown data type.
here is some logs
```
mysql> select * from test_cstick_1;
ERROR 1064 (HY000): Getting analyzing error. Detail message: Datatype of external table column [time] is not supported!.
mysql> desc test_cstick_1;
+------------+------------------+------+-------+---------+---------------+
| Field      | Type             | Null | Key   | Default | Extra         |
+------------+------------------+------+-------+---------+---------------+
| date       | DATE             | Yes  | false | NULL    |               |
| time       | UNKNOWN_TYPE     | Yes  | false | NULL    |               |
| sym        | VARCHAR(1048576) | Yes  | false | NULL    | partition key |
| prevclose  | DOUBLE           | Yes  | false | NULL    |               |
...
mysql> show create table test_cstick_1;
| test_cstick_1 | CREATE TABLE `test_cstick_1` (
  `date` date DEFAULT NULL,
  `time` UNKNOWN_TYPE DEFAULT NULL,
  `sym` varchar(1048576) DEFAULT NULL,
  `prevclose` double DEFAULT NULL,
...
```

### PR DESCRIPTION
**Pull request by leoyy0316**
Why I'm doing:
Starrocks iceberg catalog can not query time type,
What I'm doing:
1. Support iceberg time type when query iceberg catalog
2. Support insert into iceberg catalog time type table
Fixes #35231

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/formats/orc/column_reader.h [hunk 1]
```diff
@@ -145,6 +145,16 @@ class DoubleColumnReader : public PrimitiveColumnReader {
     Status get_next(orc::ColumnVectorBatch* cvb, ColumnPtr& col, size_t from, size_t size) override;
 };
 
+template <LogicalType Type>
```
Comment (Smith-Cruise) [inline be/src/formats/orc/column_reader.h:148]: Why do we need template here? time always using int64 in orc and parquet

File: be/src/formats/orc/orc_chunk_reader.cpp [hunk 2]
```diff
@@ -364,6 +364,11 @@ static Status _create_type_descriptor_by_orc(const TypeDescriptor& origin_type,
         result->len = len;
         result->precision = precision;
         result->scale = scale;
+        if (result->type == TYPE_BIGINT && orc_type->hasAttributeKey("iceberg.long-type")) {
```
Comment (Smith-Cruise) [inline be/src/formats/orc/orc_chunk_reader.cpp:367]: add some comments to explain it.

File: be/src/formats/parquet/column_converter.cpp [hunk 3]
```diff
@@ -377,6 +385,11 @@ Status ColumnConverterFactory::create_converter(const ParquetField& field, const
             *converter = std::make_unique<PrimitiveToDecimalConverter<int32_t, TYPE_DECIMAL128>>(field.scale,
                                                                                                  typeDescriptor.scale);
             break;
+        case LogicalType::TYPE_TIME: {
```
Comment (Smith-Cruise) [inline be/src/formats/parquet/column_converter.cpp:388]: why put this logic in int32?
time type must be int64

**Review by Smith-Cruise — COMMENTED** [review]
UT plz

File: be/src/formats/parquet/column_converter.cpp (hunk 3 above)
Comment (leoyy0316) [reply be/src/formats/parquet/column_converter.cpp]: @Smith-Cruise  ALL DONE

File: be/src/formats/parquet/file_writer.cpp [hunk 4]
```diff
@@ -243,10 +243,15 @@ arrow::Result<::parquet::schema::NodePtr> ParquetBuildHelper::_make_schema_node(
                                                       ::parquet::Type::BYTE_ARRAY, -1, file_column_id.field_id);
     }
     case TYPE_DATE: {
+        LOG(INFO) << "rep_type: " << rep_type;
```
Comment (Smith-Cruise) [inline be/src/formats/parquet/file_writer.cpp:246]: rm `LOG(INFO)` logs

File: be/src/formats/parquet/file_writer.cpp [hunk 5]
```diff
@@ -301,6 +306,15 @@ arrow::Result<::parquet::schema::NodePtr> ParquetBuildHelper::_make_schema_node(
         return ::parquet::schema::GroupNode::Make(name, rep_type, {key_value}, ::parquet::LogicalType::Map(),
                                                   file_column_id.field_id);
     }
+    case TYPE_TIME: {
+        LOG(INFO) << "rep_type: " << rep_type;
```
Comment (Smith-Cruise) [inline be/src/formats/parquet/file_writer.cpp:310]: ditto

File: be/src/formats/parquet/file_writer.cpp [hunk 6]
```diff
@@ -243,10 +243,15 @@ arrow::Result<::parquet::schema::NodePtr> ParquetBuildHelper::_make_schema_node(
...
     }
     case TYPE_DATE: {
+        LOG(INFO) << "rep_type: " << rep_type;
+        LOG(INFO) << "file_column_id.field_id: " << file_column_id.field_id;
+        LOG(INFO) << "name: " << name;
         return ::parquet::schema::PrimitiveNode::Make(name, rep_type, ::parquet::LogicalType::Date(),
                                                       ::parquet::Type::INT32, -1, file_column_id.field_id);
     }
     case TYPE_DATETIME: {
+        LOG(INFO) << "rep_type: " << rep_type;
```
Comment (Smith-Cruise) [inline be/src/formats/parquet/file_writer.cpp:253]: ditto

File: be/src/formats/parquet/file_writer.cpp [hunk 7]
```diff
@@ -483,6 +497,7 @@ Status AsyncFileWriter::_flush_row_group() {
 Status AsyncFileWriter::close(RuntimeState* state,
                               const std::function<void(starrocks::parquet::AsyncFileWriter*, RuntimeState*)>& cb) {
     bool ret = _executor_pool->try_offer([&, state, cb]() {
+        LOG(INFO) << ">>>>: " << 0;
```
Comment (Smith-Cruise) [inline be/src/formats/parquet/file_writer.cpp:500]: ditto

File: be/src/formats/parquet/level_builder.cpp [hunk 8]
```diff
@@ -266,6 +270,32 @@ void LevelBuilder::_write_date_column_chunk(const LevelBuilderContext& ctx, cons
...
+                                            const CallbackFunction& write_leaf_callback) {
+    const auto* data_col = get_raw_data_column<TYPE_TIME>(col);
+    const auto* null_col = get_raw_null_column(col);
+
+    // Use the rep_levels in the context from caller since node is primitive.
+    auto& rep_levels = ctx._rep_levels;
+    auto def_levels = _make_def_levels(ctx, node, null_col, col->size());
+    auto null_bitset = _make_null_bitset(ctx, null_col, col->size());
+
+    auto values = new int64_t[col->size()];
```
Comment (Smith-Cruise) [inline be/src/formats/parquet/level_builder.cpp:284]: you need delete it

File: be/test/formats/orc/orc_chunk_reader_test.cpp [hunk 9]
```diff
@@ -2276,4 +2276,79 @@ TEST_F(OrcChunkReaderTest, get_file_schema) {
...
+    Stream: column 2 section DATA start: 65 length 10
+    Encoding column 0: DIRECT
+    Encoding column 1: DIRECT_V2
+    Encoding column 2: DIRECT_V2
+
+File length: 455 bytes
+Padding length: 0 bytes
+Padding ratio: 0%
+
+{"c1":1,"c2":3723456000}
```
Comment (Smith-Cruise) [inline be/test/formats/orc/orc_chunk_reader_test.cpp:2319]: can you put more rows, and contains null value

File: be/test/formats/parquet/file_reader_test.cpp [hunk 10]
```diff
@@ -3171,4 +3171,61 @@ TEST_F(FileReaderTest, TestReadFooterCache) {
...
+    auto chunk = std::make_shared<Chunk>();
+    chunk->append_column(ColumnHelper::create_column(type_int, true), chunk->num_columns());
+    chunk->append_column(ColumnHelper::create_column(type_time, true), chunk->num_columns());
+
+    status = file_reader->get_next(&chunk);
+    ASSERT_TRUE(status.ok());
+
+    chunk->check_or_die();
+
+    EXPECT_EQ("[1, 3723]", chunk->debug_row(0));
```
Comment (Smith-Cruise) [inline be/test/formats/parquet/file_reader_test.cpp:3214]: more rows plz, and contains null value

File: test/sql/test_iceberg/T/test_iceberg_catalog_time_type [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+    "hive.metastore.uris" = "${iceberg_catalog_hive_metastore_uris}"
+);
+set catalog ice_cat_${uuid0};
+create database ice_db_${uuid0};
+use ice_db_${uuid0};
+create table ice_tbl_${uuid0} (
+  col_str int,
+  col_int time
+);
+insert into ice_tbl_${uuid0} values (1, '01-02-03'),
```
Comment (Smith-Cruise) [inline test/sql/test_iceberg/T/test_iceberg_catalog_time_type:16]: and null values

**Review by Smith-Cruise — APPROVED** [review]

**Review by stephen-shelby — DISMISSED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Review by dirtysalt — APPROVED** [review]

**Pull request merged**